cylc/flow/workflow_files.py

```diff
@@ -1491,6 +1502,51 @@ def validate_source_dir(source, flow_name):
     check_flow_file(source, logger=None)
 
 
+def get_sym_dirs(symlink_dirs: str) -> Dict[str, Dict[str, Any]]:
+    """Converts command line entered symlink dirs to a dictionary.
+
+    Args:
+        symlink_dirs (str): As entered by user on cli,
+                            e.g. "log=$DIR, share=$DIR2".
+        fill_syms (bool): If True, will fill missing
+
+    Raises:
+        WorkflowFilesError: If directory to be symlinked is not in permitted
+                            dirs: run, log, share, work, share/cycle
+
+    Returns:
+        dict: In the same form as would be returned by global config.
+            e.g. {'localhost': {'log': '$DIR',
+                                'share': '$DIR2'
+                                }
+                }
+    """
+    # Ensures the same nested dict format which is returned by the glb cfg
+    symdict: Dict[str, Dict[str, Any]] = {'localhost': {'run': None}}
+    if symlink_dirs == "":
+        return symdict
+    symlist = symlink_dirs.strip(',').split(',')
+    for pair in symlist:
+        try:
+            key, val = pair.split("=")
+            key = key.strip()
+        except ValueError:
+            raise UserInputError(
+                'There is an error in --symlink-dirs option:'
+                f' {pair}. Try entering option in the form '
+                '--symlink-dirs=\'log=$DIR, share=$DIR2, ...\''
+            )
+        if key not in ['run', 'log', 'share', 'share/cycle', 'work']:
```

Comment:
Rather than rewriting these strings,
```suggestion
        possible_symlink_dirs = WorkflowFiles.SYMLINK_DIRS.symmetric_difference(
            {'', WorkflowFiles.RUN_DIR}
        ).
        if key not in possible_symlink_dirs:
```
But it is a bit awkward due to the use of `''` instead of `'run'` in `WorkflowFiles.SYMLINK_DIRS`.

(If you do use this suggestion, instantiate `possible_symlink_dirs` outside this `for` loop rather than where I've got it)

https://docs.python.org/3.7/library/stdtypes.html#frozenset.symmetric_difference

---

Alternative:
```suggestion
        if key not in {WorkflowFiles.RUN_DIR, WorkflowFiles.LOG_DIR, ...} # etc
```